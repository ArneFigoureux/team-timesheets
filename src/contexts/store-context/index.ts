export * from "./store-context";
export * from "./store-provider";
export * from "./use-store";
