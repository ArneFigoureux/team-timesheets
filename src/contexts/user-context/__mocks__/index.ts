export const useUserStoreMock = {
    authenticatedUser: {
        email: "foobar@email.com",
        name: "Foobar",
        roles: {
            admin: true,
        },
    },
    divisionUser: {
        email: "foobar@email.com",
        name: "Foobar",
        roles: {
            admin: true,
        },
    },
} as ReturnType<typeof useUserStore>;

export const useUserStore = jest.fn().mockReturnValue(useUserStoreMock);
