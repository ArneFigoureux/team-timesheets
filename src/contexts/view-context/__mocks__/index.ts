import moment from "moment";

export const useViewStore = jest.fn().mockReturnValue({
    isDrawerOpen: true,
    moment: moment(new Date(2020, 2 ,22)),
});
