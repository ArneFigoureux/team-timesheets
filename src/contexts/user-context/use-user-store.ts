import { useStore } from "../store-context";

export const useUserStore = () => {
    const store = useStore();

    return store.user;
};
