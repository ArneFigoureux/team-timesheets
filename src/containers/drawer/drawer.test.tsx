import React from "react";
import { render } from "@testing-library/react";
import { useViewStore } from "../../contexts/view-context";
import { useUserStore } from "../../contexts/user-context";
import { Drawer } from ".";
import moment from "moment";

jest.mock("../../rules");
jest.mock("../../contexts/user-context");
jest.mock("../../contexts/view-context");

jest.mock('../../stores/router-store', () => ({
    useRouterStore: () => ({
        queryParams: {},
    }),
}));

jest.mock("../../contexts/registration-context", () => ({
    useRegistrationStore: () => ({
        registrationsGroupedByDay: [],
    }),
}));

describe("Drawer", () => {
    it("should render", () => {
        (useViewStore as jest.Mock).mockReturnValue({
            isDrawerOpen: true,
            moment: moment(new Date(2020, 2 ,22)),
        });

        (useUserStore as any).mockReturnValue({
            authenticatedUser: {
                email: "foobar@email.com",
                name: "Foobar",
            }
        });

        const { asFragment } = render(
            <Drawer />
        );

        expect(asFragment()).toMatchSnapshot();
    });
});
