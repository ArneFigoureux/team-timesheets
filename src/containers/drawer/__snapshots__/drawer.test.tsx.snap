// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer should render 1`] = `
<DocumentFragment>
  <aside
    class="mdc-drawer--open mdc-drawer--animate mdc-drawer mdc-drawer--dismissible"
  >
    <div
      class="mdc-drawer__header"
    >
      <div
        class="mdc-menu-surface--anchor"
      >
        <div
          class="division-users-menu mdc-menu-surface"
          style="transform-origin: left top; left: 0px; top: 0px; max-height: 736px;"
        >
          <ul
            class="mdc-list"
          >
            <li
              class="mdc-list-item"
              tabindex="0"
            >
              Preferences
            </li>
            <li
              class="mdc-list-item"
              tabindex="0"
            >
              Manage divisions
            </li>
            <li
              class="mdc-list-divider"
              role="separator"
            />
            <li
              class="mdc-list-item"
              tabindex="0"
            >
              <i
                aria-hidden="true"
                class="rmwc-icon rmwc-icon--ligature material-icons mdc-theme--on-surface mdc-list-item__graphic"
              >
                perm_identity
              </i>
              <span
                class="mdc-list-item__text"
              >
                Logout
              </span>
            </li>
          </ul>
        </div>
        <div
          class="account-badge"
        >
          <div
            class="account-badge__content"
          >
            <div
              class="account-badge__image"
            >
              <img
                src="https://www.gravatar.com/avatar/db2ff1aecc3bb4c910ab7289c7f920a2?d=mp&s=40"
              />
            </div>
            <div
              class="account-badge__name"
            >
              Foobar
            </div>
            <div
              class="account-badge__meta"
            >
              <i
                class="rmwc-icon rmwc-icon--ligature material-icons"
              >
                expand_more
              </i>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="mdc-drawer__content"
    >
      <div
        class="react-calendar"
      >
        <div
          class="react-calendar__navigation"
          style="display: flex;"
        >
          <button
            aria-label=""
            class="react-calendar__navigation__arrow react-calendar__navigation__prev2-button"
            type="button"
          >
            «
          </button>
          <button
            aria-label=""
            class="react-calendar__navigation__arrow react-calendar__navigation__prev-button"
            type="button"
          >
            ‹
          </button>
          <button
            aria-label=""
            class="react-calendar__navigation__label"
            style="flex-grow: 1;"
            type="button"
          >
            March 2020
          </button>
          <button
            aria-label=""
            class="react-calendar__navigation__arrow react-calendar__navigation__next-button"
            type="button"
          >
            ›
          </button>
          <button
            aria-label=""
            class="react-calendar__navigation__arrow react-calendar__navigation__next2-button"
            type="button"
          >
            »
          </button>
        </div>
        <div
          class="react-calendar__viewContainer"
        >
          <div
            class="react-calendar__month-view "
          >
            <div
              style="display: flex; align-items: flex-end;"
            >
              <div
                style="flex-grow: 1; width: 100%;"
              >
                <div
                  class="react-calendar__month-view__weekdays"
                  style="display: flex; flex-wrap: no-wrap;"
                >
                  <div
                    class="react-calendar__month-view__weekdays__weekday"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                  >
                    <abbr
                      aria-label="Monday"
                      title="Monday"
                    >
                      Mon
                    </abbr>
                  </div>
                  <div
                    class="react-calendar__month-view__weekdays__weekday"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                  >
                    <abbr
                      aria-label="Tuesday"
                      title="Tuesday"
                    >
                      Tue
                    </abbr>
                  </div>
                  <div
                    class="react-calendar__month-view__weekdays__weekday"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                  >
                    <abbr
                      aria-label="Wednesday"
                      title="Wednesday"
                    >
                      Wed
                    </abbr>
                  </div>
                  <div
                    class="react-calendar__month-view__weekdays__weekday"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                  >
                    <abbr
                      aria-label="Thursday"
                      title="Thursday"
                    >
                      Thu
                    </abbr>
                  </div>
                  <div
                    class="react-calendar__month-view__weekdays__weekday"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                  >
                    <abbr
                      aria-label="Friday"
                      title="Friday"
                    >
                      Fri
                    </abbr>
                  </div>
                  <div
                    class="react-calendar__month-view__weekdays__weekday"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                  >
                    <abbr
                      aria-label="Saturday"
                      title="Saturday"
                    >
                      Sat
                    </abbr>
                  </div>
                  <div
                    class="react-calendar__month-view__weekdays__weekday"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                  >
                    <abbr
                      aria-label="Sunday"
                      title="Sunday"
                    >
                      Sun
                    </abbr>
                  </div>
                </div>
                <div
                  class="react-calendar__month-view__days"
                  style="display: flex; flex-wrap: wrap;"
                >
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="February 24, 2020"
                    >
                      24
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="February 25, 2020"
                    >
                      25
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="February 26, 2020"
                    >
                      26
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="February 27, 2020"
                    >
                      27
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="February 28, 2020"
                    >
                      28
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend react-calendar__month-view__days__day--neighboringMonth"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="February 29, 2020"
                    >
                      29
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="March 1, 2020"
                    >
                      1
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="March 2, 2020"
                    >
                      2
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="March 3, 2020"
                    >
                      3
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="March 4, 2020"
                    >
                      4
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="March 5, 2020"
                    >
                      5
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="March 6, 2020"
                    >
                      6
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="March 7, 2020"
                    >
                      7
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="March 8, 2020"
                    >
                      8
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="March 9, 2020"
                    >
                      9
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="March 10, 2020"
                    >
                      10
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="March 11, 2020"
                    >
                      11
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="March 12, 2020"
                    >
                      12
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="March 13, 2020"
                    >
                      13
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="March 14, 2020"
                    >
                      14
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="March 15, 2020"
                    >
                      15
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="March 16, 2020"
                    >
                      16
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="March 17, 2020"
                    >
                      17
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="March 18, 2020"
                    >
                      18
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="March 19, 2020"
                    >
                      19
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="March 20, 2020"
                    >
                      20
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="March 21, 2020"
                    >
                      21
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__tile--active react-calendar__tile--rangeStart react-calendar__tile--rangeEnd react-calendar__tile--rangeBothEnds react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="March 22, 2020"
                    >
                      22
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="March 23, 2020"
                    >
                      23
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="March 24, 2020"
                    >
                      24
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="March 25, 2020"
                    >
                      25
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="March 26, 2020"
                    >
                      26
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="March 27, 2020"
                    >
                      27
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="March 28, 2020"
                    >
                      28
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="March 29, 2020"
                    >
                      29
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="March 30, 2020"
                    >
                      30
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="March 31, 2020"
                    >
                      31
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="April 1, 2020"
                    >
                      1
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="April 2, 2020"
                    >
                      2
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="April 3, 2020"
                    >
                      3
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend react-calendar__month-view__days__day--neighboringMonth"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="April 4, 2020"
                    >
                      4
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend react-calendar__month-view__days__day--neighboringMonth"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="April 5, 2020"
                    >
                      5
                    </abbr>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <ul
        class="mdc-list"
      >
        <li
          class="mdc-list-item"
          tabindex="0"
        >
          <i
            aria-hidden="true"
            class="rmwc-icon rmwc-icon--ligature material-icons mdc-theme--text-primary-on-background mdc-list-item__graphic"
          >
            today
          </i>
          <span
            class="mdc-list-item__text"
          >
            Today
          </span>
        </li>
        <li
          class="mdc-list-item"
          tabindex="0"
        >
          <i
            aria-hidden="true"
            class="rmwc-icon rmwc-icon--ligature material-icons mdc-theme--text-primary-on-background mdc-list-item__graphic"
          >
            calendar_today
          </i>
          <span
            class="mdc-list-item__text"
          >
            This month
          </span>
        </li>
        <li
          class="mdc-list-item"
          tabindex="0"
        >
          <i
            aria-hidden="true"
            class="rmwc-icon rmwc-icon--ligature material-icons mdc-theme--text-primary-on-background mdc-list-item__graphic"
          >
            favorite
          </i>
          <span
            class="mdc-list-item__text"
          >
            Favorites
          </span>
        </li>
        <li
          class="mdc-list-divider"
          role="separator"
        />
        <li
          class="mdc-list-item"
          tabindex="0"
        >
          <i
            aria-hidden="true"
            class="rmwc-icon rmwc-icon--ligature material-icons mdc-theme--text-primary-on-background mdc-list-item__graphic"
          >
            list
          </i>
          <span
            class="mdc-list-item__text"
          >
            Export
          </span>
        </li>
        <li
          class="mdc-list-item"
          tabindex="0"
        >
          <i
            aria-hidden="true"
            class="rmwc-icon rmwc-icon--ligature material-icons mdc-theme--text-primary-on-background mdc-list-item__graphic"
          >
            bar_chart
          </i>
          <span
            class="mdc-list-item__text"
          >
            Dashboard
          </span>
        </li>
        <li
          class="mdc-list-divider"
          role="separator"
        />
        <li
          class="mdc-list-item"
          tabindex="0"
        >
          <i
            aria-hidden="true"
            class="rmwc-icon rmwc-icon--ligature material-icons mdc-theme--text-primary-on-background mdc-list-item__graphic"
          >
            star_border
          </i>
          <span
            class="mdc-list-item__text"
          >
            Projects
          </span>
        </li>
        <li
          class="mdc-list-item"
          tabindex="0"
        >
          <i
            aria-hidden="true"
            class="rmwc-icon rmwc-icon--ligature material-icons mdc-theme--text-primary-on-background mdc-list-item__graphic"
          >
            settings
          </i>
          <span
            class="mdc-list-item__text"
          >
            Settings
          </span>
        </li>
        <li
          class="mdc-list-divider"
          role="separator"
        />
      </ul>
      <ul
        class="mdc-list"
        style="align-self: flex-end;"
      >
        <a
          href="https://github.com/thdk/team-timesheets/issues/new"
        >
          <li
            class="mdc-list-item"
            tabindex="0"
          >
            <svg
              aria-hidden="true"
              class="rmwc-icon rmwc-icon--component material-icons mdc-list-item__graphic"
              style="width: 24px; height: 24px;"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 0c-6.626 0-12 5.373-12 12 0 5.302 3.438 9.8 8.207 11.387.599.111.793-.261.793-.577v-2.234c-3.338.726-4.033-1.416-4.033-1.416-.546-1.387-1.333-1.756-1.333-1.756-1.089-.745.083-.729.083-.729 1.205.084 1.839 1.237 1.839 1.237 1.07 1.834 2.807 1.304 3.492.997.107-.775.418-1.305.762-1.604-2.665-.305-5.467-1.334-5.467-5.931 0-1.311.469-2.381 1.236-3.221-.124-.303-.535-1.524.117-3.176 0 0 1.008-.322 3.301 1.23.957-.266 1.983-.399 3.003-.404 1.02.005 2.047.138 3.006.404 2.291-1.552 3.297-1.23 3.297-1.23.653 1.653.242 2.874.118 3.176.77.84 1.235 1.911 1.235 3.221 0 4.609-2.807 5.624-5.479 5.921.43.372.823 1.102.823 2.222v3.293c0 .319.192.694.801.576 4.765-1.589 8.199-6.086 8.199-11.386 0-6.627-5.373-12-12-12z"
                fill="#000000"
              />
            </svg>
            Report issue
          </li>
        </a>
      </ul>
    </div>
  </aside>
</DocumentFragment>
`;
